mrs_uav_managers:

  estimation_manager:

    # loaded state estimator plugins
    state_estimators: [
      "gps_garmin",
      "gps_baro",
    ]

    initial_state_estimator: "gps_baro" # will be used as the first state estimator
    agl_height_estimator: "garmin_agl" # only slightly filtered height for checking min height (not used in control feedback)

  uav_manager:

    max_height_checking:

      enabled: false

    takeoff:

      during_takeoff:
        controller: "MpcController"
        tracker: "LandoffTracker"

      after_takeoff:
        # controller: "MpcController" # more robust
        controller: "Se3Controller" # more precise
        tracker: "MpcTracker"

mrs_uav_trackers:

  mpc_tracker:

    wiggle:
      enabled: false
      amplitude: 0.5 # [m]
      frequency: 0.2 # [Hz]

    # mrs collision avoidance
    collision_avoidance:

      enabled:           true # disabling this will stop this UAV to react to others, but it will still transmit data to others
      enabled_passively: true # disabling this will stop this uav to even transmit its data to others (only if enabled: false)
      radius: 2.0 # [m]
      correction: 3.0 # [m]

mrs_uav_trajectory_generation:

  # execution time limiter
  max_time: 0.5 # [s]

  # check and fix the max deviation between the input path and the output trajectory
  # by subsampling the original path with more waypoints
  check_trajectory_deviation:
    max_deviation: 0.20 # [m]

  # allows to neglect waypoints in the original path if the new path
  # does not violate max distance from the original path.
  # This makes for similar but sparser paths, which are easier for the
  # trajectory generator to work with.
  path_straightener:
    enabled: true
    max_deviation: 0.20 # [m]